File: src/main/scala/com/tomergabel/examples/Logging.scala
package com.tomergabel.examples

import org.slf4j.LoggerFactory


trait Logging {
  private val logger = LoggerFactory.getLogger(this.getClass)

  def warn(message: => String): Unit =
    if (logger.isWarnEnabled)
      logger.warn(message)
  
  def warn(message: => String, error: Throwable): Unit =
    if (logger.isWarnEnabled)
      logger.warn(message, error)

  def debug(message: => String): Unit =
    if (logger.isDebugEnabled)
      logger.debug(message)

  def debug(message: => String, error: Throwable): Unit =
    if (logger.isDebugEnabled)
      logger.debug(message, error)

  def info(message: => String): Unit =
    if (logger.isInfoEnabled)
      logger.info(message)

  def info(message: => String, error: Throwable): Unit =
    if (logger.isInfoEnabled)
      logger.info(message, error)

  def error(message: => String): Unit =
    if (logger.isErrorEnabled)
      logger.error(message)

  def error(message: => String, error: Throwable): Unit =
    if (logger.isErrorEnabled)
      logger.error(message, error)
}

File: src/main/java/com/tomergabel/examples/java/Logging.java
package com.tomergabel.examples.java;

import org.slf4j.Logger;

public interface Logging {
    Logger getLogger();

    default void debug(String message) {
        if (getLogger().isDebugEnabled())
            getLogger().debug(message);
    }

    default void debug(String message, Exception cause) {
        if (getLogger().isDebugEnabled())
            getLogger().debug(message, cause);
    }

    default void info(String message) {
        if (getLogger().isInfoEnabled())
            getLogger().info(message);
    }

    default void info(String message, Exception cause) {
        if (getLogger().isInfoEnabled())
            getLogger().info(message, cause);
    }

    default void warn(String message) {
        if (getLogger().isWarnEnabled())
            getLogger().warn(message);
    }

    default void warn(String message, Exception cause) {
        if (getLogger().isWarnEnabled())
            getLogger().warn(message, cause);
    }

    default void error(String message) {
        if (getLogger().isErrorEnabled())
            getLogger().error(message);
    }

    default void error(String message, Exception cause) {
        if (getLogger().isErrorEnabled())
            getLogger().error(message, cause);
    }
}
